### ISSUE TITLE
driver.js induces horizontal scrollbar on mobile

### ISSUE THREAD

**Issue by kklepper** [kamranahmedse/driver.js#286]
I made a nice, complex tour with driver.js. 

### Problem

When I reduce the clientwidth on my desktop to about 685px I can reproduce the horizontal scrollbar seen on mobile. 

#### Investigation

This happens also with all of my content removed, with a simple fake entry
```
    {
    element: "#decmaker",                                                 
    popover: {
      title: "title",
      description: "description",
      position: "bottom-center",  
      doneBtnText: "'.$ar_str['done'].'",
      closeBtnText: "'.$ar_str['close'].'",
      nextBtnText: "'.$ar_str['next'].'",
      prevBtnText: "'.$ar_str['prev'].'",
      } 
    },
```

hence I am sure that it is not my content which demands this width.

In my own css I have 

```
div#driver-popover-item{
    max-width:928px;margin-left:10px;
} 
```

which should not have this outcome, and it doesn't, as can be tested easily.

#### Proof

If I omit the tour, there is no scrollbar. So it must be something with driver.js or driver.css, but as far as I can see from the source there isn't any instruction implying this behavior.

What is going on here? Where to look at? 

#### Hint

Playing around with it, I think that most of the popups respect the given boundaries, but some do not. Of course, if there is a popup to the right in a line which ends at the right end of the viewport on mobile, but not at larger resolutions, driver makes room to the right which obviously forces the browser to introduce a horizontal scrollbar, like so:

![image](https://user-images.githubusercontent.com/6130319/117518686-0d242480-afa1-11eb-9216-c8f3f6549ed8.png)

#### Comparison

Interestingly the browser introduces the scrollbar right away, when nothing visible hints to the necessity. It obviously calculates everything in advance, visible or not, based on the code delivered. 

I noticed this behavior before with tooltips (it took me quite some time to understand this phenomenon) and wrote a workaround, so that, if necessary, their position is corrected automatically to prevent the introduction of a scrollbar.

#### Reasoning

This sounds sensible, but my fake entry proves that there must be some more to this, because there is no need for a scrollbar in this simple case.

#### Proposal

Now it is tedious to say the least to rearrange all stuff and make sure nothing will brake the barrier, but more importantly I am  not even sure if I will succeed this way. Better your code should take care of not pushing the boundaries and rearrange popups accordingly and automatically. 

This is the intention with flexboxes, if I understand correctly: the programmer should not have to care about the layout with respect to different viewports, the browser can do better because he knows.

#### Question

Is there any way to get driver.js to do this? Am I overlooking something here and it is already in place?

**Comment by kklepper** [comment]
Here is another case showing a problem with a menu line breaking:

1. viewport wide enough

![image](https://user-images.githubusercontent.com/6130319/117520259-98081d80-afa7-11eb-9ba7-3958014dd338.png)

2. viewport too narrow:

![image](https://user-images.githubusercontent.com/6130319/117519922-0946d100-afa6-11eb-8429-cfea93fdfaec.png)

**Comment by Naren-hybreeder** [comment]
@kklepper Have you found the solution for mobile devices?

**Comment by kamranahmedse** [comment]
We just released [v1.0](https://github.com/kamranahmedse/driver.js/releases/tag/1.0.3) of driver.js which has been rewritten from the ground up. This issue should no longer exist. Please have a [look at the docs](https://driverjs.com/) and do let me know if you face any issues.
